eidas_node/saml.py

```diff
@@ -0,0 +1,226 @@
+
+SAMLRequestType = TypeVar('SAMLRequestType', bound='SAMLRequest')
+SAMLResponseType = TypeVar('SAMLResponseType', bound='SAMLResponse')
+
+
+class SAMLRequest:
+    """SAML Request and its conversion from/to LightRequest."""
+
+    document = None  # type: ElementTree
+    """SAML document as an element tree."""
+    relay_state = None  # type: Optional[str]
+    """Relay state associated with the request."""
+
+    def __init__(self, document: ElementTree, relay_state: Optional[str] = None):
+        self.document = document
+        self.relay_state = relay_state
+
+    @classmethod
+    def convert_light_request(cls: Type[SAMLRequestType], light_request: LightRequest,
+                              destination: str, issued: datetime) -> SAMLRequestType:
+        """
+        Convert Light Request to SAML Request.
+
+        :param light_request: The light request to convert.
+        :param destination: A URI reference indicating the address to which this request has been sent.
+        :param issued: The UTC time instant of issue of the request.
+        :return: A SAML Request.
+        """
+        light_request.validate()
+        root_attributes = OrderedDict([
+            ('Consent', 'urn:oasis:names:tc:SAML:2.0:consent:unspecified'),  # optional, default 'unspecified'
+            ('Destination', destination),
+            ('ID', light_request.id),
+            ('IssueInstant', issued.isoformat()[:-3] + 'Z'),  # UTC
+            ('Version', '2.0'),
+            ('IsPassive', 'false'),  # optional, default false
+            ('ForceAuthn', 'true'),  # optional, default false
+        ])
+        if light_request.provider_name:
+            root_attributes['ProviderName'] = light_request.provider_name
+        root = etree.Element(Q_NAMES['saml2p:AuthnRequest'], attrib=root_attributes, nsmap=NAMESPACES)
+
+        # 1. RequestAbstractType <saml2:Issuer>:
+        if light_request.issuer:
+            SubElement(root, Q_NAMES['saml2:Issuer'],
+                       {'Format': 'urn:oasis:names:tc:SAML:2.0:nameid-format:entity'}).text = light_request.issuer
+
+        # 2. RequestAbstractType <ds:Signature> skipped
+        # 3. RequestAbstractType <saml2p:Extensions>:
+        extensions = SubElement(root, Q_NAMES['saml2p:Extensions'])
+        if light_request.sp_type:
+            SubElement(extensions, Q_NAMES['eidas:SPType']).text = light_request.sp_type.value
+        if light_request.origin_country_code:
+            SubElement(extensions, Q_NAMES['eidas:SPCountry']).text = light_request.origin_country_code
+        attributes = SubElement(extensions, Q_NAMES['eidas:RequestedAttributes'])
+        for name, values in light_request.requested_attributes.items():
+            attribute = SubElement(attributes, Q_NAMES['eidas:RequestedAttribute'], OrderedDict([
+                ('Name', name),
+                # FIXME: Is there any mapping of URI names and friendly names?
```

Comment:
https://github.com/IdentityPython/pysaml2/tree/master/src/saml2/attributemaps

Reply:
I now use `saml2.attributemaps.saml_uri`. Not sure whether I should include other maps: "adfs_v1x", "adfs_v20", "basic", "shibboleth_uri".

Reply:
I am not sure we want to bring `saml2` as a dependency... It was meant as an inspiration. It is probably too large to be used for such simple task...

Reply:
I added `attributes.py` with eIDAS natural and legal person attributes instead.